### PR TITLE
Unmark some slow tests that aren't actually slow

### PR DESCRIPTION
**Pull request by mpharrigan**
Low-hanging fruit from #1080

### CODE REVIEW

File: qualtran/bloqs/data_loading/select_swap_qrom_test.py [hunk 1]
```diff
@@ -31,15 +31,14 @@
...
-@pytest.mark.slow
 @pytest.mark.parametrize(
     "data,block_size",
     [
         pytest.param(
             data,
             block_size,
             id=f"{block_size}-data{didx}",
-            marks=pytest.mark.slow if block_size == 2 and didx == 0 else (),
+            marks=pytest.mark.slow if block_size == 1 and didx == 1 else (),
```
Comment (tanujkhattar) [inline qualtran/bloqs/data_loading/select_swap_qrom_test.py:41]: This is probably also not slow ?

**Review by tanujkhattar — APPROVED** [review]

File: qualtran/bloqs/data_loading/select_swap_qrom_test.py (hunk 1 above)
Comment (mpharrigan) [reply qualtran/bloqs/data_loading/select_swap_qrom_test.py:41]: It's >1s on my machine, but less than my personal 2s threshold for marking... but the other configurations should provide enough coverage for the `-m 'not slow'` executions of the unit tests.

File: qualtran/bloqs/hamiltonian_simulation/hamiltonian_simulation_by_gqsp_test.py [hunk 2]
```diff
@@ -83,10 +82,9 @@ def verify_hamiltonian_simulation_by_gqsp(
     assert_matrices_almost_equal(expected_top_left, actual_top_left, atol=1e-4)
 
 
-@pytest.mark.slow
 @pytest.mark.parametrize("select_bitsize", [1])
 @pytest.mark.parametrize("target_bitsize", [1, 2])
-@pytest.mark.parametrize("t", [2, 3, 5])
+@pytest.mark.parametrize("t", [2, 3, pytest.param(5, marks=pytest.mark.slow)])
```
Comment (anurudhp) [inline qualtran/bloqs/hamiltonian_simulation/hamiltonian_simulation_by_gqsp_test.py:87]: oh nice! didn't know this feature

Comment (mpharrigan) [reply qualtran/bloqs/hamiltonian_simulation/hamiltonian_simulation_by_gqsp_test.py:87]: me neither

that's also why I left in the skip from the above test that tanuj noted -- it's a nice example of how to skip particular configurations :)

**Pull request merged**